lib/elixir/lib/module/locals_tracker.ex

```diff
@@ -100,36 +100,30 @@ defmodule Module.LocalsTracker do
   end
 
   @doc """
-  Collect undefined functions based on local calls and existing definitions
+  Collect undefined functions based on local calls and existing definitions.
   """
   def collect_undefined_locals({set, bag}, all_defined) do
     undefined =
       for {pair, _, _, _} <- all_defined,
-          {local, line, _} <- out_neighbours(bag, {:local, pair}),
-          not :ets.member(set, {:def, local}),
-          do: {build_meta(line), local}
+          {local, line, macro_dispatch?} <- out_neighbours(bag, {:local, pair}),
+          reduce: [] do
```

Comment:
I would say this would be clearer as something like:

    for {pair, _, _, _} <- all_defined,
        {local, line, macro_dispatch?} <- out_neighbours(bag, {:local, pair}),
        warning = undefined_or_invalid_local(set, local, line, kind, macro_dispatch?),
        do: warning

Thoughts?

Reply:
Or perhaps even better:

```elixir
for {pair, _, _, _} <- all_defined,
    {local, line, macro_dispatch?} <- out_neighbours(bag, {:local, pair}),
    warning = undefined_or_invalid_local(set, local, kind, macro_dispatch?),
    do: {build_meta(line), local, warning} 
```

Reply:
I got carried away with `for reduce: ...` 😅 

I like your suggestion.